pkg/image/backingimage/backingimage.go

```diff
@@ -174,7 +179,12 @@ func (bib *Backend) createBackingImage(vmi *harvesterv1.VirtualMachineImage) err
 }
 
 func (bib *Backend) createStorageClass(vmi *harvesterv1.VirtualMachineImage) error {
-	if cachedSC, _ := bib.scCache.Get(util.GetImageStorageClassName(vmi)); cachedSC != nil && cachedSC.DeletionTimestamp != nil {
+	storageClassName, err := util.GetImageStorageClassName(bib.scCache, vmi)
+	if err != nil {
+		return err
+	}
+
+	if cachedSC, _ := bib.scCache.Get(storageClassName); cachedSC != nil && cachedSC.DeletionTimestamp != nil {
```

Comment:
should we return earlier if `cachedSc != nil && cachedSC.DeletionTimestamp == nil` ?

Reply:
The reconciller might run to this function might times.

Either use the existing one, or compare parameters and if not same, update it or delete/create new one and log related message/event. WDYT?